compiler/runtime/OMRRSSReport.cpp

```diff
@@ -0,0 +1,377 @@
+          itemList.isEmpty())
+         {
+         TR_VerboseLog::writeLineLocked(TR_Vlog_PERF, "RSS: Resident page at addr %p has no RSS items", addr);
+         }
+      else if (resident == 1)
+         {
+         ListIterator<OMR::RSSItem> it(&itemList);
+
+         size_t totalItemSize = 0;
+         int32_t countItems = 0;
+         uint8_t *prevEnd = (addr == start) ? addr : (uint8_t *)(((uintptr_t)addr / pageSize) * pageSize);  // current page start
+         OMR::RSSItem *prevItem = NULL;
+
+         uint64_t pageDebugCount = 0;
+
+         // print RSS items
+         for (OMR::RSSItem *item = it.getFirst(); item; item = it.getNext())
+            {
+            uint64_t itemDebugCount = 0;
+
+            if (item->_counters)
+               {
+               ListIterator<TR::DebugCounterAggregation> aggrIt(item->_counters);
+
+               // print debug counters
+               for (TR::DebugCounterAggregation *aggr = aggrIt.getFirst(); aggr; aggr = aggrIt.getNext())
+                  {
+                  uint64_t count = aggr->getCount();
+
+                  itemDebugCount += count;
+                  pageDebugCount += count;
+                  if (count > 0) aggr->printCounters(false /*printZeroCounters*/);
+                  }
+               }
+
+            int32_t gap = item->_addr - prevEnd;
+
+            TR_ASSERT_FATAL(gap >= 0, "Item at addr %p in page %p is out of order, prevAddr=%p", item->_addr, addr,
+                            prevItem->_addr);
+
+            if (gap != 0)
+               {
+               TR_VerboseLog::writeLineLocked(TR_Vlog_PERF, "RSS: gap at addr=%p size=%zu in %s region", prevEnd, gap, rssRegion->_name);
+               }
+
+            totalItemSize += item->_size + gap;
+
+            TR_VerboseLog::writeLineLocked(TR_Vlog_PERF, "RSS item at addr=%p size=%zu itemDebugCount=%zu %s",
+                                           item->_addr, item->_size, itemDebugCount, OMR::RSSItem::itemNames[item->_type]);
+
+            countItems++;
+            prevEnd = item->_addr + item->_size;
+            prevItem = item;
+            }
+
+         TR_VerboseLog::writeLineLocked(TR_Vlog_PERF, "RSS: Page at addr %p has %zu bytes of %d items pageDebugCount=%zu",
+                                        addr, totalItemSize, countItems, pageDebugCount);
+
+         TR_ASSERT_FATAL(totalItemSize <= pageSize, "Total size of items within page %p > page size\n", addr);
+         }
+
+      pageCount++;
+      }
+#endif
+
+   return rss;
+   }
+
+void
+OMR::RSSReport::printTitle()
+   {
+   int32_t regionNum = 0;
+   char rssLine[MAX_RSS_LINE];
+
+   ListIterator<OMR::RSSRegion> it(&_regions);
+   for (OMR::RSSRegion *rssRegion = it.getFirst(); rssRegion; rssRegion = it.getNext())
+      {
+      if ((regionNum * REGION_PRINT_WIDTH + 1) < MAX_RSS_LINE)
+         {
+         sprintf(rssLine + regionNum * REGION_PRINT_WIDTH, "%*.*s", REGION_PRINT_WIDTH, REGION_PRINT_WIDTH, rssRegion->_name);
```

Comment:
`sprintf` has been causing grief with newer build compilers due to potential overflow issues being flagged as errors.  I think it's best not to introduce new uses and just go with `snprintf` instead.

See #6863 and a comment from Devin on some ways of consolidating snprintf behaviour across compilers: https://github.com/eclipse/omr/issues/6863#issuecomment-1382937072

Reply:
Replaced sprintf with TR::snprintfTrunc.